homeassistant/components/homekit/accessories.py

```diff
@@ -137,6 +138,24 @@ def update_state(self, new_state):
         """
         raise NotImplementedError()
 
+    def call_service(self, acc_domain, service, params, logbook_message=None):
+        """Sent event for logbook that a change was initiated from homekit.
+
+        Then call the service to change the state.
+        This method is to be called from methods that change states using
+        self.call_service(DOMAIN, service, params, logbook_message).
+        """
+        if logbook_message is None:
+            logbook_message = service
```

Comment:
Construct the `event_data` dict beforehand and only add the `value` if given.

Reply:
Based on changes done do not have this anymore. Just firing event with the 4 parameters.

Reply:
The `value` parameter will still be optional, but I guess we could just pass `None` as well if we do the check and message construction in the `logbook` component.